pkg/cri/resource-manager/cache/cache.go

```diff
@@ -1110,6 +1131,84 @@ func (cch *cache) setEntry(key string, ptr, obj interface{}) error {
 	return nil
 }
 
+// checkPerm checks ownership and permissions if a file or directory exists.
+func (cch *cache) checkPerm(what, path string, isDir bool, p *permissions) (bool, error) {
+	if isDir {
+		what += " directory"
+	}
+
+	info, err := os.Stat(path)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return true, cacheError("failed to os.Stat() %s %q: %v", what, path, err)
+		}
+		return false, nil
+	}
+
+	// check expected file type
+	if isDir {
+		if !info.IsDir() {
+			return true, cacheError("%s %q exists, but is not a directory", what, path)
+		}
+	} else {
+		if info.Mode()&os.ModeType != 0 {
+			return true, cacheError("%s %q exists, but is not a regular file", what, path)
+		}
+	}
+
+	// check expected root/root ownership
+	stat, ok := info.Sys().(*syscall.Stat_t)
+	if !ok || stat == nil {
+		return true, cacheError("failed to get ownership of %s %q: can't get syscall.Stat_t",
+			what, path)
+	}
+	if stat.Uid != 0 || stat.Gid != 0 {
```

Comment:
Do we want to strictly enforce this? I think it is possible to run the container runtime as rootless, for example. Or maybe run cri-resmgr as non-root when doing introspection only, for example. WDYT?

Reply:
Hmm... I considered checking against geteuid()/getegid() then went with this. Perhaps both because I was a bit too much in SDL mode. Let's drop the user/group ownership checks altogether.